src/log_batch.rs

```diff
@@ -527,6 +559,12 @@ enum BufState {
     /// # Invariants
     /// LOG_BATCH_HEADER_LEN <= buf.len()
     Sealed(usize, usize),
+    /// Buffer contains header, entries and footer; ready to be written. The
+    /// footer may be signed with extra information depending on the format
+    /// version.
+    /// # Content
+    /// (header_offset, entries_len)
+    Encoded(usize, usize),
```

Comment:
Encoded and Sealed, not Sealed and Encoded. Sealed means the data has become immutable.